src/crypto_backends/openssl.rs

```diff
@@ -100,6 +87,28 @@ impl LocalKeyPair for OpenSSLLocalKeyPair {
             pub_key_point.to_bytes(&GROUP_P256, PointConversionForm::UNCOMPRESSED, &mut bn_ctx)?;
         Ok(uncompressed)
     }
+
+    fn from_raw_components(components: &EcKeyComponents) -> Result<Self> {
+        let d = BigNum::from_slice(&components.private_key())?;
+        let mut bn_ctx = BigNumContext::new()?;
+        let ec_point = EcPoint::from_bytes(&GROUP_P256, &components.public_key(), &mut bn_ctx)?;
```

Comment:
This also brushes against the idea that maybe the curve type should be in a `Default` that gets brought in rather than being hard coded here.

Reply:
Aye. We hard-code P-256 in various places here because it's hard-coded in the ece standard, but I wonder what it would look like to forward-evolve this for some future standard that uses a different curve. One option would be would be to tag it as an additional field on the `EcKeyComponents`. Another could be to make the hard-coding really explicit like we do for aes-gcm, and name all the functions like `generate_p256_ephemeral_keypair()`, `type RemotePublicKeyP256` and so-on.

Reply:
I've added `curve` to `EcKeyComponents`.